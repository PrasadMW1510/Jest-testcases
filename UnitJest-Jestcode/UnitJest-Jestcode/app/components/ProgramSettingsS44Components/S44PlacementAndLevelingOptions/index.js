/**
 *
 * S44PlacementAndLevelingOptions
 *
 */

export { default } from './S44PlacementAndLevelingOptions';
