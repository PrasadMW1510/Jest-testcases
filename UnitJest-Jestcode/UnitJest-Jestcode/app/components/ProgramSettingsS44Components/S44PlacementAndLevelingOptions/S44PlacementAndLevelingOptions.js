/**
 *
 * S44PlacementAndLevelingOptions
 *
 */
import PropTypes from 'prop-types';
import React from 'react';

import SettingsFourStateCheckbox from 'components/SettingsFourStateCheckbox';

import '../ProgramSettingsS44Components.scss';

function S44PlacementAndLevelingOptions({
  settingsObj,
  callBackFunc,
  isHalfHeight,
  isHalfWidth,
  isS44NG,
}) {
  const getHeightClassName = () =>
    isHalfHeight ? 's44ng-settings__section--half-height' : 's44ng-settings__section--full-height';

  const getWidthClassName = () =>
    isHalfWidth ? 's44ng-settings__section--half-width' : 's44ng-settings__section--full-width';

  const handleChangeAutoPlacement = () => {
    callBackFunc({ ...settingsObj, auto_placement: ['1'], initial_placement: ['0'] });
  };

  const handleChangeEnableFastTrack = e => {
    callBackFunc({ ...settingsObj, enable_fasttrack: [e.target.checked ? '1' : '0'] });
  };

  const handleChangeInitialValue = e => {
    callBackFunc({ ...settingsObj, initial_placement: [e.target.value.toString()] });
  };

  const handleChangeSetInitialRadio = () => {
    callBackFunc({ ...settingsObj, auto_placement: ['0'] });
  };

  const createSetInitialDropDown = () =>
    Array.from(Array(25).keys()).map((currentValue, index) => (
      <option key={currentValue} value={index + 1}>
        {index + 1}
      </option>
    ));

  const renderHasNotStartedWorking = () => (
    <div className="s44ng-settings__initial-settings">
      <div className="s44ng-settings__radio">
        <input
          id="autoPlacement"
          type="radio"
          checked={settingsObj.auto_placement[0] === '1'}
          onChange={handleChangeAutoPlacement}
        />
        <span className="s44ng-settings__radio-text">
          Enable automatic placement based on initial Phonics Inventory results
        </span>
      </div>
      <div className="s44ng-settings__radio">
        <input
          id="setInitial"
          type="radio"
          checked={settingsObj.auto_placement[0] === '0'}
          onChange={handleChangeSetInitialRadio}
        />
        <span className="s44ng-settings__radio-text">Set initial series to:</span>
        <select
          onChange={handleChangeInitialValue}
          value={settingsObj.initial_placement[0]}
          disabled={settingsObj.auto_placement[0] === '1'}
          className={`s44ng-settings__select ${
            settingsObj.auto_placement[0] === '1' ? 's44ng-settings__select--disabled' : ''
          }`}
        >
          <option value={0}>-</option>
          {createSetInitialDropDown()}
        </select>
      </div>
    </div>
  );

  const renderHasStartedWorking = () => (
    <div className="s44ng-settings__has-started-working">
      {`Once student has started System 44 ${isS44NG ? 'NG' : ''} the`}
      <br />
      “Placement and Leveling” options are unavailable
    </div>
  );

  return (
    <div className={`s44ng-settings__section ${getHeightClassName()} ${getWidthClassName()}`}>
      <div className="s44ng-settings__section-title">Placement and Leveling Options</div>
      <div className="s44ng-settings__section-content">
        {settingsObj.has_started_working[0] === 'false' && renderHasNotStartedWorking()}
        {settingsObj.has_started_working[0] === 'true' && renderHasStartedWorking()}
        <SettingsFourStateCheckbox
          checkboxText="Enable Fast-Track between series"
          currentCheckboxValue={settingsObj.enable_fasttrack[0]}
          handleChangeCheckboxValue={handleChangeEnableFastTrack}
        />
      </div>
    </div>
  );
}

S44PlacementAndLevelingOptions.defaultProps = {
  isHalfHeight: true,
  isHalfWidth: true,
  isS44NG: false,
};

S44PlacementAndLevelingOptions.propTypes = {
  callBackFunc: PropTypes.func.isRequired,
  settingsObj: PropTypes.object.isRequired,
  isHalfHeight: PropTypes.bool,
  isHalfWidth: PropTypes.bool,
  isS44NG: PropTypes.bool,
};

export default S44PlacementAndLevelingOptions;
