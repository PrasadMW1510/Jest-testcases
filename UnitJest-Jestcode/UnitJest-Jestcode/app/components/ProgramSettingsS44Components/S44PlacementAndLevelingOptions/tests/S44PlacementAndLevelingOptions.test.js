import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

import S44PlacementAndLevelingOptions from '../index';

describe('<S44PlacementAndLevelingOptions />', () => {
  let wrapper = null;

  let mockCallBackFunc = null;
  let mockSettingObj = null;

  const fakeTrueEvent = {
    target: {
      checked: true,
    },
  };

  const fakeFalseEvent = {
    target: {
      checked: false,
    },
  };

  const testEnableFastTrack = () => {
    let enableFastTrackCheckbox = null;

    beforeEach(() => {
      enableFastTrackCheckbox = wrapper.find(
        'SettingsFourStateCheckbox[checkboxText="Enable Fast-Track between series"]'
      );
    });

    it('handles true click', () => {
      enableFastTrackCheckbox.prop('handleChangeCheckboxValue')(fakeTrueEvent);

      mockSettingObj.enable_fasttrack = ['1'];
      expect(mockCallBackFunc).toHaveBeenCalledWith(mockSettingObj);
    });

    it('handles false click', () => {
      enableFastTrackCheckbox.prop('handleChangeCheckboxValue')(fakeTrueEvent);
      enableFastTrackCheckbox.prop('handleChangeCheckboxValue')(fakeFalseEvent);

      mockSettingObj.enable_fasttrack = ['0'];
      expect(mockCallBackFunc).toHaveBeenCalledWith(mockSettingObj);
    });
  };

  beforeEach(() => {
    mockCallBackFunc = jest.fn();
  });

  describe('student hasnt started working', () => {
    describe('auto_placement is 0', () => {
      beforeEach(() => {
        mockSettingObj = {
          has_started_working: ['false'],
          auto_placement: ['0'],
          initial_placement: ['0'],
          enable_fasttrack: ['0'],
          spanish_support: ['0'],
          captioning: ['0'],
          writing_enabled: ['0'],
        };

        wrapper = shallow(
          <S44PlacementAndLevelingOptions
            callBackFunc={mockCallBackFunc}
            settingsObj={mockSettingObj}
          />
        );
      });

      it('Expect to render correctly', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
      });

      describe('Initial settings radio buttons', () => {
        let autoPlacementRadioButton = null;
        let setInitialRadioButton = null;

        beforeEach(() => {
          autoPlacementRadioButton = wrapper.find('input[id="autoPlacement"]');
          setInitialRadioButton = wrapper.find('input[id="setInitial"]');
        });

        describe('auto placement radio button', () => {
          it('auto_placement is set to 0', () => {
            autoPlacementRadioButton.simulate('change');

            mockSettingObj.auto_placement = ['1'];
            mockSettingObj.initial_placement = ['0'];
            expect(mockCallBackFunc).toHaveBeenCalledWith(mockSettingObj);
          });
        });

        describe('set initial radio button', () => {
          it('auto_placement is set to 0', () => {
            autoPlacementRadioButton.simulate('change');
            setInitialRadioButton.simulate('change');

            mockSettingObj.auto_placement = ['0'];
            expect(mockCallBackFunc).toHaveBeenCalledWith(mockSettingObj);
          });
        });
      });

      it('Initial value select on change', () => {
        const initialValueSelect = wrapper.find('select');
        initialValueSelect.simulate('change', { target: { value: '17' } });

        mockSettingObj.initial_placement = ['17'];
        expect(mockCallBackFunc).toHaveBeenCalledWith(mockSettingObj);
      });

      describe('Enable Fast Track checkbox', () => {
        testEnableFastTrack();
      });
    });

    describe('auto_placement is 1', () => {
      beforeEach(() => {
        mockSettingObj = {
          has_started_working: ['false'],
          auto_placement: ['1'],
          initial_placement: ['0'],
          enable_fasttrack: ['0'],
          spanish_support: ['0'],
          captioning: ['0'],
          writing_enabled: ['0'],
        };

        wrapper = shallow(
          <S44PlacementAndLevelingOptions
            callBackFunc={mockCallBackFunc}
            settingsObj={mockSettingObj}
          />
        );
      });

      it('Expect to render correctly', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
      });

      describe('Initial settings radio buttons', () => {
        let autoPlacementRadioButton = null;
        let setInitialRadioButton = null;

        beforeEach(() => {
          autoPlacementRadioButton = wrapper.find('input[id="autoPlacement"]');
          setInitialRadioButton = wrapper.find('input[id="setInitial"]');
        });

        describe('auto placement radio button', () => {
          it('auto_placement is set to 0', () => {
            autoPlacementRadioButton.simulate('change');

            mockSettingObj.auto_placement = ['1'];
            mockSettingObj.initial_placement = ['0'];
            expect(mockCallBackFunc).toHaveBeenCalledWith(mockSettingObj);
          });
        });

        describe('set initial radio button', () => {
          it('auto_placement is set to 0', () => {
            autoPlacementRadioButton.simulate('change');
            setInitialRadioButton.simulate('change');

            mockSettingObj.auto_placement = ['0'];
            expect(mockCallBackFunc).toHaveBeenCalledWith(mockSettingObj);
          });
        });
      });

      it('Initial value select on change', () => {
        const initialValueSelect = wrapper.find('select');
        initialValueSelect.simulate('change', { target: { value: '17' } });

        mockSettingObj.initial_placement = ['17'];
        expect(mockCallBackFunc).toHaveBeenCalledWith(mockSettingObj);
      });

      describe('Enable Fast Track checkbox', () => {
        testEnableFastTrack();
      });
    });
  });

  describe('student has started working', () => {
    describe('isS44NG is false', () => {
      beforeEach(() => {
        mockSettingObj = {
          has_started_working: ['true'],
          auto_placement: ['0'],
          initial_placement: ['0'],
          enable_fasttrack: ['0'],
          spanish_support: ['0'],
          captioning: ['0'],
          writing_enabled: ['0'],
        };

        wrapper = shallow(
          <S44PlacementAndLevelingOptions
            callBackFunc={mockCallBackFunc}
            settingsObj={mockSettingObj}
            isHalfHeight={false}
            isHalfWidth={false}
          />
        );
      });

      it('Expect to render correctly', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
      });

      describe('Enable Fast Track checkbox', () => {
        testEnableFastTrack();
      });
    });

    describe('isS44NG is true', () => {
      beforeEach(() => {
        mockSettingObj = {
          has_started_working: ['true'],
          auto_placement: ['0'],
          initial_placement: ['0'],
          enable_fasttrack: ['0'],
          spanish_support: ['0'],
          captioning: ['0'],
          writing_enabled: ['0'],
        };

        wrapper = shallow(
          <S44PlacementAndLevelingOptions
            callBackFunc={mockCallBackFunc}
            settingsObj={mockSettingObj}
            isHalfHeight={false}
            isHalfWidth={false}
            isS44NG
          />
        );
      });

      it('Expect to render correctly', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
      });

      describe('Enable Fast Track checkbox', () => {
        testEnableFastTrack();
      });
    });
  });
});
