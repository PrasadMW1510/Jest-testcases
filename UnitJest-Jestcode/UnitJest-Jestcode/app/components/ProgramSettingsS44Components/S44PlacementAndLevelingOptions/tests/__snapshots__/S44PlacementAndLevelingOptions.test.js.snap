// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<S44PlacementAndLevelingOptions /> student has started working isS44NG is false Expect to render correctly 1`] = `
<div
  className="s44ng-settings__section s44ng-settings__section--full-height s44ng-settings__section--full-width"
>
  <div
    className="s44ng-settings__section-title"
  >
    Placement and Leveling Options
  </div>
  <div
    className="s44ng-settings__section-content"
  >
    <div
      className="s44ng-settings__has-started-working"
    >
      Once student has started System 44  the
      <br />
      “Placement and Leveling” options are unavailable
    </div>
    <SettingsFourStateCheckbox
      checkboxClass=""
      checkboxName="settingsFourStateCheckbox"
      checkboxText="Enable Fast-Track between series"
      currentCheckboxValue="0"
      disabled={false}
      handleChangeCheckboxValue={[Function]}
      statusMap={
        Object {
          "checkedDisabled": Array [],
          "checkedEnabled": Array [
            "1",
          ],
          "mixedDisabled": Array [],
          "mixedEnabled": Array [
            "-2",
            "2",
          ],
          "uncheckedDisabled": Array [
            "-1",
          ],
          "uncheckedEnabled": Array [
            "0",
          ],
        }
      }
      textClass=""
    />
  </div>
</div>
`;

exports[`<S44PlacementAndLevelingOptions /> student has started working isS44NG is true Expect to render correctly 1`] = `
<div
  className="s44ng-settings__section s44ng-settings__section--full-height s44ng-settings__section--full-width"
>
  <div
    className="s44ng-settings__section-title"
  >
    Placement and Leveling Options
  </div>
  <div
    className="s44ng-settings__section-content"
  >
    <div
      className="s44ng-settings__has-started-working"
    >
      Once student has started System 44 NG the
      <br />
      “Placement and Leveling” options are unavailable
    </div>
    <SettingsFourStateCheckbox
      checkboxClass=""
      checkboxName="settingsFourStateCheckbox"
      checkboxText="Enable Fast-Track between series"
      currentCheckboxValue="0"
      disabled={false}
      handleChangeCheckboxValue={[Function]}
      statusMap={
        Object {
          "checkedDisabled": Array [],
          "checkedEnabled": Array [
            "1",
          ],
          "mixedDisabled": Array [],
          "mixedEnabled": Array [
            "-2",
            "2",
          ],
          "uncheckedDisabled": Array [
            "-1",
          ],
          "uncheckedEnabled": Array [
            "0",
          ],
        }
      }
      textClass=""
    />
  </div>
</div>
`;

exports[`<S44PlacementAndLevelingOptions /> student hasnt started working auto_placement is 0 Expect to render correctly 1`] = `
<div
  className="s44ng-settings__section s44ng-settings__section--half-height s44ng-settings__section--half-width"
>
  <div
    className="s44ng-settings__section-title"
  >
    Placement and Leveling Options
  </div>
  <div
    className="s44ng-settings__section-content"
  >
    <div
      className="s44ng-settings__initial-settings"
    >
      <div
        className="s44ng-settings__radio"
      >
        <input
          checked={false}
          id="autoPlacement"
          onChange={[Function]}
          type="radio"
        />
        <span
          className="s44ng-settings__radio-text"
        >
          Enable automatic placement based on initial Phonics Inventory results
        </span>
      </div>
      <div
        className="s44ng-settings__radio"
      >
        <input
          checked={true}
          id="setInitial"
          onChange={[Function]}
          type="radio"
        />
        <span
          className="s44ng-settings__radio-text"
        >
          Set initial series to:
        </span>
        <select
          className="s44ng-settings__select "
          disabled={false}
          onChange={[Function]}
          value="0"
        >
          <option
            value={0}
          >
            -
          </option>
          <option
            key="0"
            value={1}
          >
            1
          </option>
          <option
            key="1"
            value={2}
          >
            2
          </option>
          <option
            key="2"
            value={3}
          >
            3
          </option>
          <option
            key="3"
            value={4}
          >
            4
          </option>
          <option
            key="4"
            value={5}
          >
            5
          </option>
          <option
            key="5"
            value={6}
          >
            6
          </option>
          <option
            key="6"
            value={7}
          >
            7
          </option>
          <option
            key="7"
            value={8}
          >
            8
          </option>
          <option
            key="8"
            value={9}
          >
            9
          </option>
          <option
            key="9"
            value={10}
          >
            10
          </option>
          <option
            key="10"
            value={11}
          >
            11
          </option>
          <option
            key="11"
            value={12}
          >
            12
          </option>
          <option
            key="12"
            value={13}
          >
            13
          </option>
          <option
            key="13"
            value={14}
          >
            14
          </option>
          <option
            key="14"
            value={15}
          >
            15
          </option>
          <option
            key="15"
            value={16}
          >
            16
          </option>
          <option
            key="16"
            value={17}
          >
            17
          </option>
          <option
            key="17"
            value={18}
          >
            18
          </option>
          <option
            key="18"
            value={19}
          >
            19
          </option>
          <option
            key="19"
            value={20}
          >
            20
          </option>
          <option
            key="20"
            value={21}
          >
            21
          </option>
          <option
            key="21"
            value={22}
          >
            22
          </option>
          <option
            key="22"
            value={23}
          >
            23
          </option>
          <option
            key="23"
            value={24}
          >
            24
          </option>
          <option
            key="24"
            value={25}
          >
            25
          </option>
        </select>
      </div>
    </div>
    <SettingsFourStateCheckbox
      checkboxClass=""
      checkboxName="settingsFourStateCheckbox"
      checkboxText="Enable Fast-Track between series"
      currentCheckboxValue="0"
      disabled={false}
      handleChangeCheckboxValue={[Function]}
      statusMap={
        Object {
          "checkedDisabled": Array [],
          "checkedEnabled": Array [
            "1",
          ],
          "mixedDisabled": Array [],
          "mixedEnabled": Array [
            "-2",
            "2",
          ],
          "uncheckedDisabled": Array [
            "-1",
          ],
          "uncheckedEnabled": Array [
            "0",
          ],
        }
      }
      textClass=""
    />
  </div>
</div>
`;

exports[`<S44PlacementAndLevelingOptions /> student hasnt started working auto_placement is 1 Expect to render correctly 1`] = `
<div
  className="s44ng-settings__section s44ng-settings__section--half-height s44ng-settings__section--half-width"
>
  <div
    className="s44ng-settings__section-title"
  >
    Placement and Leveling Options
  </div>
  <div
    className="s44ng-settings__section-content"
  >
    <div
      className="s44ng-settings__initial-settings"
    >
      <div
        className="s44ng-settings__radio"
      >
        <input
          checked={true}
          id="autoPlacement"
          onChange={[Function]}
          type="radio"
        />
        <span
          className="s44ng-settings__radio-text"
        >
          Enable automatic placement based on initial Phonics Inventory results
        </span>
      </div>
      <div
        className="s44ng-settings__radio"
      >
        <input
          checked={false}
          id="setInitial"
          onChange={[Function]}
          type="radio"
        />
        <span
          className="s44ng-settings__radio-text"
        >
          Set initial series to:
        </span>
        <select
          className="s44ng-settings__select s44ng-settings__select--disabled"
          disabled={true}
          onChange={[Function]}
          value="0"
        >
          <option
            value={0}
          >
            -
          </option>
          <option
            key="0"
            value={1}
          >
            1
          </option>
          <option
            key="1"
            value={2}
          >
            2
          </option>
          <option
            key="2"
            value={3}
          >
            3
          </option>
          <option
            key="3"
            value={4}
          >
            4
          </option>
          <option
            key="4"
            value={5}
          >
            5
          </option>
          <option
            key="5"
            value={6}
          >
            6
          </option>
          <option
            key="6"
            value={7}
          >
            7
          </option>
          <option
            key="7"
            value={8}
          >
            8
          </option>
          <option
            key="8"
            value={9}
          >
            9
          </option>
          <option
            key="9"
            value={10}
          >
            10
          </option>
          <option
            key="10"
            value={11}
          >
            11
          </option>
          <option
            key="11"
            value={12}
          >
            12
          </option>
          <option
            key="12"
            value={13}
          >
            13
          </option>
          <option
            key="13"
            value={14}
          >
            14
          </option>
          <option
            key="14"
            value={15}
          >
            15
          </option>
          <option
            key="15"
            value={16}
          >
            16
          </option>
          <option
            key="16"
            value={17}
          >
            17
          </option>
          <option
            key="17"
            value={18}
          >
            18
          </option>
          <option
            key="18"
            value={19}
          >
            19
          </option>
          <option
            key="19"
            value={20}
          >
            20
          </option>
          <option
            key="20"
            value={21}
          >
            21
          </option>
          <option
            key="21"
            value={22}
          >
            22
          </option>
          <option
            key="22"
            value={23}
          >
            23
          </option>
          <option
            key="23"
            value={24}
          >
            24
          </option>
          <option
            key="24"
            value={25}
          >
            25
          </option>
        </select>
      </div>
    </div>
    <SettingsFourStateCheckbox
      checkboxClass=""
      checkboxName="settingsFourStateCheckbox"
      checkboxText="Enable Fast-Track between series"
      currentCheckboxValue="0"
      disabled={false}
      handleChangeCheckboxValue={[Function]}
      statusMap={
        Object {
          "checkedDisabled": Array [],
          "checkedEnabled": Array [
            "1",
          ],
          "mixedDisabled": Array [],
          "mixedEnabled": Array [
            "-2",
            "2",
          ],
          "uncheckedDisabled": Array [
            "-1",
          ],
          "uncheckedEnabled": Array [
            "0",
          ],
        }
      }
      textClass=""
    />
  </div>
</div>
`;
